src/typing/typeload.ml

```diff
@@ -345,8 +345,8 @@ let rec load_instance' ctx (t,p) allow_no_params =
 				| TPExpr e ->
 					let name = (match fst e with
 						| EConst (String(s,_)) -> "S" ^ s
-						| EConst (Int i) -> "I" ^ i
-						| EConst (Float f) -> "F" ^ f
+						| EConst (Int (i,_)) -> "I" ^ i
```

Comment:
This may lead to misses for types like `MyType<123i32>` and `MyType<123i64>`

Reply:
Good point, we can just append the suffix here.

Reply:
Actually this makes me realize that we might have to sanitize the suffixes somewhere if they aren't typed as expressions. I think we could sneak in invalid strings via macros otherwise.

Reply:
ah, yes, I think I see what you mean. `macro : MyType<123i40>` allows the i40 suffix into the type parameter (I'm currently ignoring the suffix in that part of the reify since I wasn't sure when it was called until now).

I'll look into not ignoring the suffixes in this case and adding some sort of check, not sure if there are any other places you could add invalid suffixes as well.

Reply:
I've added a check in macro api decoding to ensure the suffixes of any created numeric constants are valid. Not sure if there's a better place to put these checks so its not done both in the typer and at macro decoding.